### ISSUE TITLE
Functions with default argument values

### ISSUE THREAD

**Issue by rejhgadellaa** [LennyN95/weboutline#4]
The Weboutline package seems to have issues with functions that have default argument values set:

```
function doSomething(options={}) {
  // do something
}
```

This function will not show up in the outline panel.

**Comment by kevingrabher** [comment]
Changing the regex for php functions from

`/(function\s?([\w\$]*)\s*\([\w\s\$,]*)\)|(class\s([\w\$]*)[^{]*{)|(\/\/[\s]*@(.*))/g`
to
`/(function\s?([\w\$]*)\s*\(.*)\)|(class\s([\w\$]*)[^{]*{)|(\/\/[\s]*@(.*))/g`

( [\w\s\$,]* to .* )
would match functions with default arguments as well.

However, my regex experience is rather poor, so I'm not sure if this would lead to other unexpected behaviour.

**Comment by LennyN95** [comment]
@rejhgadellaa thx for creating this issue. Back then, default values for arguments were not possible (at least I didn't know them). 

@kevingrabher thx for your aproach. I think just allowing the equal sign in the function parameters should already do the trick like so: `(function\s?([\w\$]*)\s*\([\w\s\$\=,]*)\)|(class\s([\w\$]*)[^{]*{)|(\/\/[\s]*@(.*))`. I'll test that and update the repo then. 

If you need it added more quickly, feel free to add a pull request or let me know ;)

**Comment by kevingrabher** [comment]
Hi Lenny,

This works for me if the default value is an int for example, but doesn't trigger on objects, arrays and strings.
We'd need to add a few more chars like `{} [] " '`
`(function\s?([\w\$]*)\s*\([\w\s\$\=\{\}\'\"\[\],]*)\)|(class\s([\w\$]*)[^{]*{)|(\/\/[\s]*@(.*))`

**Comment by LennyN95** [comment]
Good point. But in that case I think there is a bit more work required. It's a few years ago
So I do not remember without looking into the code
But in case that the parameters are displayed in the outline: with eg. object structures as optional parameter values, this can get long and ugly real quick.

**Comment by kevingrabher** [comment]
Perhaps an option _could_ be added to only show the function name without parameters, but I don't really see that as a necessity. I find it to be convenient to see the whole thing. On big files my experience is you'll need to use the search function anyhow.
